src/main/kotlin/com/jetbrains/snakecharm/lang/psi/SmkReferenceExpression.kt

```diff
@@ -48,6 +51,9 @@ class SmkReferenceExpression(node: ASTNode): PyElementImpl(node), PsiNamedElemen
             return variants.toTypedArray()
         }
 
+        override fun handleElementRename(newElementName: String): PsiElement =
+                element.let { (it as PsiNamedElement).setName(newElementName) }
```

Comment:
We don't need a cast here, you get "element" from constructor, so please use ther element with proper type.